src/plugins/core/merge.ts

```diff
@@ -254,11 +254,14 @@ export class MergePlugin extends CorePlugin<MergeState> implements MergeState {
   }
 
   getMainCellPosition(position: CellPosition): CellPosition {
-    if (!this.isInMerge(position)) {
-      return position;
-    }
-    const mergeTopLeftPos = this.getMerge(position)!.topLeft;
-    return { sheetId: position.sheetId, col: mergeTopLeftPos.col, row: mergeTopLeftPos.row };
+    const mergeZone = this.getMerge(position);
+    return mergeZone
+      ? {
+          sheetId: position.sheetId,
+          col: mergeZone?.left as HeaderIndex,
+          row: mergeZone?.top as HeaderIndex,
```

Comment:
casting is useless here (and should always be avoided when possible).
```suggestion
          sheetId: position.sheetId,
          col: mergeZone.left,
          row: mergeZone.top,
```